Xamarin.Forms.Xaml/CollectionExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+
+namespace Xamarin.Forms.Xaml
+{
+	public static class CollectionExtensions
+	{
+		public static object EnsureCollectionInitialized(this BindableObject bindable, BindableProperty property)
+		{
```

Comment:
it's not up to the Xaml to create collection if not explicitly declared. if you want to add to a collection, that collection has to exists (can be lazily created in the property getter).
we're doing this at other places

Reply:
Fixed for BindableObject - the collections are now initialized lazily when the value is retrieved.